data/tokenizer.py

```diff
@@ -0,0 +1,19 @@
+class Tokenizer:
+    def __init__(
+        self,
+        keys: list[str],
+    ):
+        self.keys = keys
+
+    def __call__(self, record: dict) -> list[str]:
+        samples = []
+
+        for idx in range(record[self.keys[0]].size):
+            sample = str()
+
+            for key in self.keys:
+                sample += f"{record[key][idx]}-"
+
+            samples.append(sample[:-1])
+
+        return samples
```

Comment:
```suggestion
    def __call__(self, df: pd.DataFrame) -> list[str]:
        samples = []
        
        for it, row in df.iterrows():
            sample = "-".join([row[key] for key in self.keys])
            samples.append(samples)
        return samples
```

It's good to think about midi structures as data frames. If you don't want to use pandas here for speed reasons, you should still try to simplify this method and try to make it look/read more like this pandas version :)